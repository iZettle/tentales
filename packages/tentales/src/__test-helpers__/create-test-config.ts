/* tslint:disable no-empty */

export function createTestConfig() {
  return {
    port: 12345,
    public: true,
    auth: {
      serverSecret: "test-secret",
    },
    reactComponentsDirectory: "",
    services: {
      renderer: {
        host: "this",
        path: "/tt/renderer",
      },
      data: {
        host: "this",
        path: "/tt/data",
      },
      editor: {
        host: "this",
        path: "/tt/editor",
      },
    },
    hooks: {
      middlewares: {
        beforeRenderMiddleware: [() => {}, () => {}],
        rendererService: [() => {}],
      },
    },
  }
}
