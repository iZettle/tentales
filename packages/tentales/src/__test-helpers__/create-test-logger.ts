import { Log } from "tentales-log"

export const createTestLogger = (): Log => ({
  silly: jest.fn(),
  verbose: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
})
