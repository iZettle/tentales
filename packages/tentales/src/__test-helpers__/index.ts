export { createTestConfig } from "./create-test-config"
export { createTestMwConfig } from "./create-test-mw-config"
export { createTestMw } from "./create-test-mw"
