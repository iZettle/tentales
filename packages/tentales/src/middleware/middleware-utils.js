function getHookMiddlewaresFromConfig(config) {
  if (!config.hooks || !config.hooks.middlewares) {
    return []
  }

  return Object.keys(config.hooks.middlewares).reduce((acc, name) => {
    acc.push([name, config.hooks.middlewares[name]])
    return acc
  }, [])
}

module.exports = { getHookMiddlewaresFromConfig }
