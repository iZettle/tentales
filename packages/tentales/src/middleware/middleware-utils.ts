import { Config, MiddlewareDefinition, Middleware } from "../types"

function getMiddlewaresAtHook(
  hookName: string,
  hookMiddlewares: any,
): Middleware[] {
  return hookMiddlewares[hookName] as Middleware[]
}

export function getHookMiddlewaresFromConfig(
  config: Config,
): MiddlewareDefinition[] {
  if (!config.hooks || !config.hooks.middlewares) {
    return []
  }

  const hookMiddlewares = config.hooks.middlewares

  return Object.keys(config.hooks.middlewares).reduce(
    (acc, hookName) => {
      const mwDefinition: MiddlewareDefinition = [
        hookName,
        getMiddlewaresAtHook(hookName, hookMiddlewares),
      ]
      acc.push(mwDefinition)
      return acc
    },
    [] as MiddlewareDefinition[],
  )
}
