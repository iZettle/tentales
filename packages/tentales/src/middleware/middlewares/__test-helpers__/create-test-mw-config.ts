import { always } from "ramda"

export function createTestMwConfig() {
  const ttConfig = {
    port: 12345,
    public: true,
    auth: {
      serverSecret: "",
    },
    reactComponentsDirectory: "",
    services: {
      renderer: {
        host: "this",
        path: "/tt/renderer",
      },
      data: {
        host: "this",
        path: "/tt/data",
      },
      editor: {
        host: "this",
        path: "/tt/editor",
      },
    },
    hooks: {
      middlewares: {},
    },
  }

  return {
    services: {
      renderer: async () => await {},
      data: async () => await {},
      editor: async () => await {},
    },
    log: {
      silly: always(null),
      verbose: always(null),
      info: always(null),
      warn: always(null),
      error: always(null),
    },
    config: ttConfig,
  }
}
