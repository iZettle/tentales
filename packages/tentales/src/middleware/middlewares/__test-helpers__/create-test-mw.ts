/* tslint:disable no-any no-empty */

export function createTestMw({
  body = "Next body",
  contentType = null,
  status = null,
  fn = (ctx: any) => {},
} = {}) {
  return async function nextMw(ctx: any, next: any) {
    ctx.body = body

    const header = {}

    if (contentType) {
      ctx.response.type = contentType
    }

    if (status) {
      ctx.status = status
    }

    if (header) {
      ctx.set(header)
    }

    fn(ctx)

    await next()
  }
}
