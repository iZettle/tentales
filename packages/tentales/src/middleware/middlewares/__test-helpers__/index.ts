export { createTestMwConfig } from "./create-test-mw-config"
export { createTestMw } from "./create-test-mw"
