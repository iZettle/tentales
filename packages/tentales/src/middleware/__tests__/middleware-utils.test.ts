import { createTestConfig } from "../../__test-helpers__"
import { createConfigHooks } from "../middleware-utils"

test("It create config hooks", async () => {
  const resp = createConfigHooks(createTestConfig())
  expect(resp).toMatchSnapshot()
})
