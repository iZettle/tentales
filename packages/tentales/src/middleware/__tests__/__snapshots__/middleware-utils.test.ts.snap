// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It create config hooks 1`] = `Array []`;
